CONTAINERFILE_GUIDE.md

```diff
@@ -0,0 +1,203 @@
+        self.wfile.write(b"Hello World!")
+
+if __name__ == "__main__":
+    server_address = ('', 8080)  # Serve on all addresses, port 8080
+    httpd = HTTPServer(server_address, HelloWorldHandler)
+    print("Starting httpd...")
+    httpd.serve_forever()
+```
+
+2. Save the systemd unit file to `helloworld.service`
+
+```sh
+[Unit]
+Description=Hello World HTTP Server
+Wants=network-online.target
+After=network-online.target
+
+[Install]
+WantedBy=default.target
+
+[Service]
+ExecStart=/usr/bin/python3 /usr/bin/helloworld.py
+Restart=on-failure
+```
+
+3. Create the `Containerfile`:
+
+```Dockerfile
+FROM quay.io/centos-bootc/fedora-bootc:eln
+
+# Add the systemd service
+ADD helloworld.service /usr/lib/systemd/system/helloworld.service
+
+# Equivilant "systemctl enable helloworld"
+# will be implemented in the future
+RUN ln -s \
+/usr/lib/systemd/system/helloworld.service \
+/usr/lib/systemd/system/default.target.wants/helloworld.service
+
+# Add the helloworld example
+ADD helloworld.py /usr/bin/helloworld.py
+
+# Change the root password so I can login
+RUN echo "root:root" | chpasswd
+```
+
+4. Follow the [usage steps](https://github.com/containers/podman-desktop-extension-bootc?tab=readme-ov-file#usage) for your container.
+
+5. Confirm that it is working. Once you boot your container, login and `curl localhost:8080`. Your web server is running!
+
+
+### Immutable container / Podman Quadlet method
+
+The **preferred** and immutable way is with [podman quadlets](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html) which is a systemd auto-generation integration built into every OS that has `podman` installed. This adds the ability to create a `[Container]` section. 
+
+You can learn how to create a full `[Container]` section on the [podman site](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html).
+
+Below we will have three files:
+- `helloworld.container`: a podman quadlet systemd unit file that runs an http server
+- `Containerfile`: our Containerfile that builds our OS and adds the systemd unit file
+
+1. Save the systemd podman quadlet file to `helloworld.container`:
+
+```sh
+[Unit]
+Description=Hello World HTTP server
+Wants=network-online.target
+After=network-online.target
+
+# More information is available on the Podman website
+[Container]
+Image=docker.io/library/caddy
+# TODO
+# NOT WORKING AND UNSURE WHY AT THE MOMENT
+PublishPort=8080:8080
+
+[Install]
+WantedBy=default.target
+
+[Service]
+# Restart service when sleep finishes
+Restart=always
+# Extend Timeout to allow time to pull the image
+TimeoutStartSec=900
+```
+
+There are other non-official tools such as [podlet](https://github.com/k9withabone/podlet) that can automatically generate the systemd-compatible file from a `podman run` command which can help generate a file quicker.
+
+2. Create the `Containerfile`:
+
+```Dockerfile
+FROM quay.io/centos-bootc/fedora-bootc:eln
+
+COPY storage.conf /etc/containers/storage.conf
```

Comment:
Is the storage.conf necessary? Is there an example storage.conf?

Reply:
nm, found it below :)

Reply:
Unfortunately yes as of right now :(.. As there is only a few persistent folders for the container OS.

I cannot find the issue, but I believe this "hack" will be changed in the future. Same goes for the whole 'systemctl enable" commands not working within a container.

Reply:
I have been running bootc images without the storage.conf addition - seems to work fine 🤷

Reply:
fyi - the additional stores in storage.conf is configured by default now in fedora-bootc:eln - however it is not yet in centos-bootc:stream9 - I learned the hard way, ran out of space when pulling images in a centos-bootc system.